### ISSUE TITLE
Event(s) around \Mage_Adminhtml_Block_System_Config_Form_Fieldset::render

### ISSUE THREAD

**Issue by pquerner** [OpenMage/magento-lts#3376]
### Description (*)

It would be great if we had any events around `\Mage_Adminhtml_Block_System_Config_Form_Fieldset::render`, so we can have more control over a single field being rendered.

This is coming because we use https://github.com/avstudnitz/AvS_ScopeHint and would like to add more fields to it. By default it only handles text-ish fields okay. SerializedArray fields dont work at all. 
After a hacky process I got it to work (not without changing the module AND my classes using the SerializedArray); however I think it would have been much easier if I had the chance of changing (ie. transportObject) the item being rendered there.

**Comment by fballiano** [comment]
actually https://github.com/avstudnitz/AvS_ScopeHint would be a nice addition to the core itself...

**Comment by pquerner** [comment]
True. I am currently here patching it so it works with SerializedArray forms;

<img width="860" alt="image" src="https://github.com/OpenMage/magento-lts/assets/4364044/81045b05-2766-4240-986f-587a40bef850">

It works, but its very hacky. An event at the right places would have saved me some time.

**Comment by fballiano** [comment]
I'm pro adding both of the things ;-)

**Comment by pquerner** [comment]
My idea was (when it all is cleaned up) pushing it to AvS_ScopeHint. (wouldnt get merged there anyways I think, Andreas moved to M2 long ago and probably doesnt touch M1 anymore).

**Comment by pquerner** [comment]
Should we ask AvS about porting his modules changes to the core itself? I am not sure. Also there are some open issues left. If we were to port it to the core itself, port it 1:1 at first and then fix issues, right?

**Comment by fballiano** [comment]
mmm it is true that there are quite some issues: https://github.com/avstudnitz/AvS_ScopeHint/issues, some of them seem quite important, like https://github.com/avstudnitz/AvS_ScopeHint/issues/27 and probably should be fixed in the porting.

For sure if AvS would do it, it would be great.

**Comment by pquerner** [comment]
No nono, I am asking generally about taking opensource modules which now seem possible to merge into the codebase. How that should be handled.

(Also, how to not break instances which have these modules installed, or is that a willing BC to take)

I dont think AvS is taking care of that, but it is his copyright afterall.

We need a battleplan for this. :D And answers, like whats happening to the copyright. If we can touch it, or what.

**Comment by pquerner** [comment]
Can a moderator move this to a discussion (I dont know how to do it) please?
I'd like to work out a plan for this with you guys, so I can work on it.

Topics to discuss (at hand)
- How and if we are allowed to take 3rd party opensource M1 modules and put its functionality in the core
 (take as-is with minor adaptions (cleaner code, ready for php8 and so forth) - or complete rewrite? (aka when we dont have to use observers, or introduce a specific new event etc))
- How and if we should display difference about values inbetween the store-views. (Same as AvS_ScopeHint? A little cleaner? Less detail? SerializedArray from the Database shouldnt be diffed imho. Just if they differ the text "Change detected" (or something) should be enough) -- only show for admins? add acl role? etc.pp
